src/core/model/Document.cpp

```diff
@@ -152,17 +156,23 @@ auto Document::createSaveFilename(DocumentType type, const std::string& defaultS
 
     const char* format = wildcardString.empty() ? defaultSaveName.c_str() : wildcardString.c_str();
 
+    // Todo (cpp20): use <format>
+    std::ostringstream ss;
+    ss.imbue(std::locale(""));
     time_t curtime = time(nullptr);
-    char stime[128];
-    strftime(stime, sizeof(stime), format, localtime(&curtime));
-
-    // Remove the extension, file format is handled by the filter combo box
-    fs::path p = stime;
+    ss << std::put_time(localtime(&curtime), format);
+    auto filename = ss.str();
+    // Todo (cpp20): use <ranges>
+    for (auto& c : filename) {
+        if (forbiddenChars.find(c) != std::string::npos) {
+            c = '_';
```

Comment:
Here we must also check, if a character is greater 127 and or any other invisible character like \n/\t

Reply:
This should go into util/PathUtil, I think.

Reply:
The thing is, that this function parses the input from a user, we will never need it elsewhere. It also does not work on paths itself, only on filenames. Gtk already does something similar, it just ignores the filename, if wrong.